app/src/main/java/ch/epfl/qedit/backend/location/MockLocService.java

```diff
@@ -0,0 +1,61 @@
+package ch.epfl.qedit.backend.location;
+
+import android.content.Context;
+import android.location.Location;
+import android.location.LocationListener;
+import java.util.HashSet;
+import java.util.Set;
+
+/** This class is used to write tests for the parts of the app that need the location service. */
+public class MockLocService implements LocationService {
+    /** This is the id of the mock location provider */
+    public static final String LOCATION_PROVIDER = "mock_provider";
+
+    // This is the set of listeners that need to be updated.
+    private final Set<LocationListener> listeners;
+
+    // This indicates whether the user has permissions to access this service.
+    // Note that this is for test purposes. It allows to test how the UI reacts when the user
+    // doesn't have the access to the location.
+    private boolean permission;
+
+    public MockLocService(Context context) {
+        listeners = new HashSet<>();
+        permission = false;
+    }
+
+    @Override
+    public boolean subscribe(LocationListener listener, int interval) {
+        // In the mock, we will not take into account the interval, because the goal
```

Comment:
You don't use `interval` ?

Reply:
No, but I added it for the real location service, so that the location is not updated too quickly. But in the mock, I want to be able to set the location wherever I want.